app/src/main/java/app/revanced/manager/ui/viewmodel/PatcherViewModel.kt

```diff
@@ -151,29 +104,66 @@ class InstallerViewModel(
                         app.toast(app.getString(R.string.install_app_fail, extra))
                     }
                 }
-
-                UninstallService.APP_UNINSTALL_ACTION -> {
-                }
             }
         }
     }
 
     init {
+        // TODO: navigate away when system-initiated process death is detected because it is not possible to recover from it.
+
         ContextCompat.registerReceiver(app, installBroadcastReceiver, IntentFilter().apply {
             addAction(InstallService.APP_INSTALL_ACTION)
-            addAction(UninstallService.APP_UNINSTALL_ACTION)
+            //addAction(UninstallService.APP_UNINSTALL_ACTION)
         }, ContextCompat.RECEIVER_NOT_EXPORTED)
+
+        viewModelScope.launch {
+            installedApp = installedAppRepository.get(packageName)
+        }
+
+        val (selectedApp, patches) = input
+
+        _steps = MutableStateFlow(
+            PatcherProgressManager.generateSteps(
+                app,
+                patches.flatMap { it.value },
+                selectedApp
+            )
+        )
+
+        patcherWorkerId =
+            workerRepository.launchExpedited<PatcherWorker, PatcherWorker.Args>(
+                "patching", PatcherWorker.Args(
+                    input.selectedApp,
+                    outputFile.path,
+                    input.selectedPatches,
+                    input.options,
+                    _steps,
+                    logger,
+                    setInputFile = { inputFile = it }
+                )
+            )
     }
 
-    fun exportLogs(context: Context) {
+    val steps = _steps.map { steps -> steps.groupBy { it.category } }
+
+    val patcherSucceeded =
+        workManager.getWorkInfoByIdLiveData(patcherWorkerId).map { workInfo: WorkInfo ->
+            when (workInfo.state) {
+                WorkInfo.State.SUCCEEDED -> true
+                WorkInfo.State.FAILED -> false
+                else -> null
+            }
+        }
+
+    fun exportLogs() {
         val sendIntent: Intent = Intent().apply {
             action = Intent.ACTION_SEND
             putExtra(Intent.EXTRA_TEXT, logger.export())
             type = "text/plain"
         }
 
         val shareIntent = Intent.createChooser(sendIntent, null)
-        context.startActivity(shareIntent)
+        app.startActivity(shareIntent)
```

Comment:
Have you tested this? I think I was using `context` instead of `app` for a reason